app/backends/workflow-orchestration.ts

```diff
@@ -590,8 +584,30 @@ export async function updateWorkItem(req: HarmonyRequest, res: Response): Promis
       }
     }
   });
-  if (!responded) {
-    // If we haven't returned an error to the caller already return a success with no body
-    res.status(204).send();
-  }
+}
+
+/**
+ * Update a work item from a service response. This function stores the update without further
+ * processing and then responds quickly. Processing the update is handled asynchronously 
+ * (see `handleWorkItemUpdate`)
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ * @returns Resolves when the request is complete
+ */
+export async function updateWorkItem(req: HarmonyRequest, res: Response): Promise<void> {
+  const { id } = req.params;
+  const { status, hits, results, scrollID, errorMessage } = req.body;
+  const totalGranulesSize = req.body.totalGranulesSize ? parseFloat(req.body.totalGranulesSize) : 0;
+
+  const update = new WorkItemUpdate(
+    { workItemID: id, status, hits, serializedResults: JSON.stringify(results), scrollID, errorMessage, totalGranulesSize },
+  );
+
+  // asynchronously handle the update
```

Comment:
Give us more info on the comment on why we are doing this - what problem are we looking to address? What happens if the asynchronous update fails, and why is it ok that we ignore it?

Reply:
Done.